test/python/test_instructions.py

```diff
@@ -270,6 +271,100 @@ def test_get_all_hwm(
         f"Expected test hmw {test_hwm} but got {received_test_hwm}"
 
 
+def build_preattestation(op_level, op_round, chain_id):
+    """Build a preattestation."""
+    return Preattestation(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_attestation(op_level, op_round, chain_id):
+    """Build a attestation."""
+    return Attestation(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_attestation_dal(op_level, op_round, chain_id):
+    """Build a attestation_dal."""
+    return AttestationDal(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_block(level, current_round, chain_id):
+    """Build a block."""
+    return Block(
+        header=BlockHeader(
+            level=level,
+            fitness=Fitness(current_round=current_round)
+        )
+    ).forge(chain_id=chain_id)
+
+PARAMETERS = [
+    (build_attestation,    (0, 0), build_preattestation,  (0, 1), True ),
+    (build_block,          (0, 1), build_attestation_dal, (1, 0), True ),
+    (build_block,          (0, 1), build_attestation_dal, (0, 0), False),
+    (build_attestation,    (1, 0), build_preattestation,  (0, 1), False),
+] + [
+    (build_preattestation,  (0, 0), build_preattestation,  (0, 0), False),
+    (build_preattestation,  (0, 0), build_block,           (0, 0), False),
+    (build_preattestation,  (0, 0), build_attestation,     (0, 0), True ),
+    (build_preattestation,  (0, 0), build_attestation_dal, (0, 0), True ),
+    (build_attestation,     (0, 0), build_preattestation,  (0, 0), False),
+    (build_attestation,     (0, 0), build_attestation,     (0, 0), False),
+    (build_attestation,     (0, 0), build_attestation_dal, (0, 0), False),
+    (build_attestation,     (0, 0), build_block,           (0, 0), False),
+    (build_attestation_dal, (0, 0), build_preattestation,  (0, 0), False),
+    (build_attestation_dal, (0, 0), build_attestation,     (0, 0), False),
+    (build_attestation_dal, (0, 0), build_attestation_dal, (0, 0), False),
+    (build_attestation_dal, (0, 0), build_block,           (0, 0), False),
+    (build_block,           (1, 1), build_preattestation,  (1, 1), True ),
+    (build_block,           (1, 1), build_attestation,     (1, 1), True ),
+    (build_block,           (1, 1), build_attestation_dal, (1, 1), True ),
```

Comment:
Do we expect to be able to sign a consensus operation after signing a block at the same level and in the same round?

Reply:
No. After build_attestation , the block is signed. So if build_attestation_dal comes with same level and round, its double baking attempt and should be rejected.

Reply:
I've had a few discussions with some people, and it seems that a baker can bake a block and then attest it at the same level and same round.
In fact, this happens when the baker attests his own block.
Double baking only occurs when a baker bakes two different blocks (and double attestation only occurs when an attester attests two different blocks) at the same level and in the same round.
If a baker bakes a block and attests another, this is not considered dangerous and will not be punished.

Do you agree?

Reply:
It seems i misunderstood. Yes you can bake the block, pre-attest it and attest it , all having same level and round.

Reply:
every attestation attempt needs to have a distinct level,round combo. Thats the only limitation, whether its same block or different thats not the criteria.

Reply:
Are we on the same page?